python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -336,40 +344,65 @@ def step(self):
                     # Note: If we have GPU nccl support, we can use all_to_all to
                     # sync information here
                     gather_list = list(
-                        th.empty([self._world_size], dtype=th.int64).chunk(
-                            self._world_size
-                        )
-                    )
-                    alltoall_cpu(
-                        self._rank,
-                        self._world_size,
-                        gather_list,
-                        idx_split_size,
+                        th.empty(
+                            [self._world_size], dtype=th.int64, device=device
+                        ).chunk(self._world_size)
                     )
+                    if th.distributed.get_backend() == "nccl":
```

Comment:
how about wrapping these if-else into a function like `dgl.all_to_all()` which is more brief in code or you need to add such logic in multiple places.

Reply:
Added.